### PR TITLE
feat: use micronaut data annotations

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc [hunk 1]
```diff
@@ -80,6 +76,7 @@ The previous domain has a `OneToMany` relationship with the domain `Book`.
 
 source:domain/Book[]
 
+TIP: You could use JPA annotations instead by including the following `compileOnly` scoped dependency `jakarta.persistence:jakarta.persistence-api`.
```
Comment (graemerocher) [inline guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc]: Add a link to https://micronaut-projects.github.io/micronaut-data/latest/guide/#sqlAnnotations which describes the supported annotations and which ones are supported

**Review by graemerocher — COMMENTED** [review]
I have some questions since the example is confusing to me.

* The guide creates ApplicationConfiguration and an implementation configuration properties but this is not used anywhere, so why is this necessary for the guide?
* A Book entity is created but then not used anywhere in the example, nor is association handling demonstrated or tested anywhere. Why is it there?
* Why create command objects like GenreSaveCommand which only have a single property? Is there some functional reason for this?
Once I have answers to these questions I will send a PR to the branch you created. Thanks!

**Comment by graemerocher** [comment]
Any feedback?

**Comment by sdelamo** [comment]
working on your suggestions now.

**Comment by sdelamo** [comment]
@graemerocher :

> * The guide creates ApplicationConfiguration and an implementation configuration properties but this is not used anywhere, so why is this necessary for the guide?

In the [Micronaut SQL Hibernate JPA guide](http://localhost/guides/micronaut-data-jdbc-repository-gradle-java.html#data-source-configuration) we use `application.max` to show the user how we use configuration to limit the maximum number of results. E.g: 

`query.setMaxResults(args.getMax().orElseGet(applicationConfiguration::getMax));`

In this guide, it makes more sense to tell the user that he can use `micronaut.data.pageable.default-page-size` instead. 

I have removed `ApplicationConfiguration`.

**Comment by sdelamo** [comment]
> * A Book entity is created but then not used anywhere in the example, nor is association handling demonstrated or tested anywhere. Why is it there?

I have removed the `Book` Entity.  It was there originally because we had such domain composition in a previous guide. And I think it is good to show some common one-to-many mapping. We can do that in a different guide.

**Comment by sdelamo** [comment]
> * Why create command objects like GenreSaveCommand which only have a single property? Is there some functional reason for this?

I personally prefer to encapsulate the HTTP Requests' body in a POJO. Do you want me to use method parameters instead of POJOs?

File: guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc (hunk 1 above)
Comment (sdelamo) [reply guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc]: I've added the link.

**Comment by graemerocher** [comment]
> I personally prefer to encapsulate the HTTP Requests' body in a POJO. Do you want me to use method parameters instead of POJOs?

This pattern makes sense if you have several properties, but in the case where there is only one it is just noise and more verbosity

**Comment by sdelamo** [comment]
> > I personally prefer to encapsulate the HTTP Requests' body in a POJO. Do you want me to use method parameters instead of POJOs?
> 
> This pattern makes sense if you have several properties, but in the case where there is only one it is just noise and more verbosity

I've deleted `GenreCreateCommand` which had only one field. Now, the method signature looks like: 

`public HttpResponse<Genre> save(@Body("name") @NotBlank String name) {`

**Comment by sdelamo** [comment]
@graemerocher can I merge this?

**Review by graemerocher — APPROVED** [review]

**Pull request merged**